src/metabase/server/middleware/settings_cache.clj

```diff
@@ -0,0 +1,45 @@
+(ns metabase.server.middleware.settings-cache
+  "Ring middleware to check for settings updates based on a cookie value."
+  (:require
+   [metabase.settings.core :as setting]
+   [metabase.util.log :as log]
+   [ring.util.response :as response]))
+
+(def ^:private settings-last-updated-cookie-name "metabase.SETTINGS_LAST_UPDATED")
+
+(defn- check-and-update-settings-cache
+  "Check if the cookie timestamp is newer than our cached timestamp, and if so, update the cache."
+  [request]
+  (when-let [cookie-timestamp (get-in request [:cookies settings-last-updated-cookie-name :value])]
+    (let [last-known-update   (setting/cache-last-updated-at)]
```

Comment:
nit: you could `when-let` with `last-known-update` too

Reply:
i updated this to an `if-let` because i need to return the response in that case. but you correct